setup-app.py

```diff
@@ -24,3 +81,113 @@
 
 # Now the app can be started
 call(["cf", "start", appname])
+
+# Tail the application log
+print("Parsing log files.")
+end_token = "cf stop %s" % appname  # Seeing this in the log means certs done
+log_pipe = Popen("cf logs %s --recent" % appname, shell=True,
+                 stdout=PIPE, stderr=PIPE)
+log_lines = log_pipe.stdout.readlines()
+print("Waiting for certs (could take several minutes)")
+seconds_waited = 0
+MAX_WAIT_SECONDS = 600
+while end_token not in ''.join(log_lines)\
+        and seconds_waited < MAX_WAIT_SECONDS:
+    # Keep checking the logs for cert readiness
+    print("Certs not ready yet, retrying in 5 seconds.")
+    time.sleep(5)
+    seconds_waited = seconds_waited + 5
+    log_pipe = Popen("cf logs %s --recent" % appname, shell=True,
+                     stdout=PIPE, stderr=PIPE)
+    log_lines = log_pipe.stdout.readlines()
+
+# If no certs in log after 10 minutes, exit and warn user
+if seconds_waited >= MAX_WAIT_SECONDS:
+    print("\n\nIt has been %d minutes without seeing certificates issued"
+          % (MAX_WAIT_SECONDS/60)
+          + " in the log. Something probably went wrong. Please check"
+          + " the output of `cf logs %s --recent`" % appname
+          + " for more information.\n\nExiting.")
+    sys.exit(1)
+
+# Figure out which domain name to look for
+primary_domain = settings['domains'][0]['domain']
+
+# Now that certs should be ready, parse for the commands to fetch them
+cmds = []
+for line in log_lines:
+    if ("cf files %s" % appname) in line and primary_domain in line:
+        cmds.append(line)
+
+# Preprocess and transform commands
+for idx, cmd in enumerate(cmds):
+    # Break each command into chunks and ignore everything before
+    # 'cf files ...'
+    parts = [s.strip() for s in cmd.split(' ') if s != ''][3:]
+    # Join the parts back together. This is necessary so that
+    # it's easy to find all of the unique commands
+    cmds[idx] = ' '.join(parts)
+
+# Toss them in a set to keep only unique commands, then convert
+# to a list again so that they can be broken into sublists
+cmds = list(set(cmds))
+
+# Extract the parts of each command that are of interest
+cmds = [cmd.split(' ') for cmd in cmds]
+for idx, cmd in enumerate(cmds):
```

Comment:
Yes, we actually pull 4 files here. Let's Encrypt generates:

- `cert.pem`, your SSL certificate
- `chain.pem`, the intermediate signing certificate for your new SSL
- `privkey.pem`, the private key used to generate your SSL (keep this secure or your SSL is useless)
- `fullchain.pem`, some other intermediate certificate. I honestly don't know what it does.